In [1]:
import sys
import platform
from math import sqrt, pi

print('Python version: ', sys.version)
print('Running on: ', sys.platform)

Python version:  3.6.3 |Anaconda, Inc.| (default, Oct 27 2017, 12:14:30) 
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]
Running on:  darwin


**Given**: A steel bar with a cross-section of 130 mm<sup>2</sup> and a length of 50.8 mm is axially loaded in tension with a force of 57.0 kN. The modulus of elasticity of the steel is 210 GPa.

**Find**: What is the axial stress in the bar?

**Solution**: 

F = 57.0 kN

F = 5700 N

A<sub>0</sub> = 130 mm<sup>2</sup>

&sigma; = ?

$$ \sigma = \frac{F}{A_0} $$



In [2]:
F = 57000
A0 = 130
stress = F/A0
print(stress)

438.46153846153845


**Find**: What is the resulting strain?

**Solution**: 

&sigma; = 438.46 MPa

E = 210 GPa

E = 210000 MPa

&epsilon; = ?

$$ E = \frac{\Delta\sigma}{\Delta\epsilon} $$

In [3]:
E = 210000
# E = stress/strain
strain = stress/E
print(strain)

0.0020879120879120877


**Find**: What is the change in length of the bar?

**Solution**:

&epsilon; = 0.0020879

L<sub>0</sub> = 50.8 mm

&Delta;L = ?

$$ \epsilon = \frac{\Delta L}{L_0} $$

$$ \Delta L = \epsilon L_0 $$


In [4]:
L0=50.8
deltaL =strain*L0
print(deltaL)

0.10606593406593405


**Find**: What is the final length of the bar?

**Solution**:

L<sub>0</sub> = 50.8 mm

&Delta;L = 0.10607 mm

L<sub>f</sub> = ?

$$ L_f = L_0 +\Delta L $$

In [5]:
Lf = L0 + deltaL
print(Lf)

50.90606593406593


#2. **Given**: An aluminum rod will be loaded in tension with a force of 350 kip. 

**Find**: What is the minimum diameter required to have a factor of safety of 1.9 agaist yield, if the yield stress is 40 ksi?

**Solution**:

SF = 1.9

&sigma;<sub>y</sub> = 40 ksi

d = ?

In [6]:
SF =1.9
stress_y = 40
stress_app = stress_y/SF
print('Stress: ',stress_app)
F = 350
A0 = F/stress_app
print('Cross-sectional Area: ', A0)
d = sqrt(4*A0/pi)
print('Diameter, d = ', d)

Stress:  21.05263157894737
Cross-sectional Area:  16.625
Diameter, d =  4.600826820390231


#3. A composite rod is axially loaded. It is ridigidly mounted at the wall and a single load <b>P</b> of 50 kN is applied at the end.

| Material | Area (mm<sup>2</sup>) | Length (mm) | Elastic Modulus, E (GPa) | Yield Strength, &sigma;<sub>y</sub> (MPa) |
| --- | --- | --- | --- | --- |
|Aluminum     | 400 | 600 | 70.0| 240 |
|Brass   | 300 | 800 | 105| 410 |

a. What is the stress in the Aluminum?

In [7]:
A_Al=400
F = 50000
stress_Al= F/A_Al
print(stress_Al)

125.0


b. What is the stress in the Brass?

In [8]:
A_Br = 300
stress_Br = F/A_Br
print(stress_Br)

166.66666666666666


c. What is the final length of the entire composite rod once the load <b>P</b> is applied?

In [9]:
E_Al = 70000
strain_Al = stress_Al/E_Al
print(strain_Al)

0.0017857142857142857


In [10]:
L0_Al = 600
dL_Al = strain_Al*L0_Al
print(dL_Al)

1.0714285714285714


In [11]:
E_Br = 105000
strain_Br = stress_Br/E_Br
print(strain_Br)

0.0015873015873015873


In [12]:
L0_Br = 800
dL_Br = strain_Br*L0_Br
print(dL_Br)

1.2698412698412698


In [13]:
Lf = L0_Al + dL_Al + L0_Br + dL_Br
print(Lf)

1402.3412698412696


d. What is the factor of safety against yeild for the brass?

In [14]:
stress_y_Br = 410
SF_Br = stress_y_Br / stress_Br
print(SF_Br)

2.46
